adi:ad7134: Add support for ad7134

jansunil opened the pull request:
1. Add pyadi driver for ad405x family
2. Add example script for data capture with AD4052
3. Add documentation

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Hardware:
* OS:

# Documentation

If this is a new feature or example please mention or link any documentation. All new hardware interface classes require documentation.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

jansunil commented:
Changelog: Fixed pylint and build errors

jansunil commented:
@tfcollins  - Could you please review these changes when you get a chance?

tfcollins commented on supported_parts.md:
```diff
@@ -70,6 +70,8 @@
 - AD6676
 - AD7124
+- AD7134
+- AD4134
```
Please put these in numeric order

tfcollins commented on adi/ad7134.py:
```diff
@@ -0,0 +1,131 @@
+# Copyright (C) 2024 Analog Devices, Inc.
+#
+# SPDX short identifier: ADIBSD
+# Copyright (C) 2024 Analog Devices, Inc.
```
Full license not needed. Just the SPDX

tfcollins commented on adi/ad7134.py:
```diff
@@ -0,0 +1,131 @@
+    """ AD7134 ADC """
+
+    _complex_data = False
+    channel = []  # type: ignore
```
Should be called "channels". Needs to be reset/initialized in the constructor similar to the issue in the related PR

tfcollins reviewed (changes requested):
Only minor issues to fix

jansunil commented:
Changelog: Addressed review comments
1) Corrected the license header
2) reword channel->channels
3) Ordered AD4134 in numeric order in the supported_parts.md

jansunil commented:
@tfcollins - Do these changes look good now?
